docs/templates/base.html

```diff
@@ -0,0 +1,144 @@
+{% extends "adidoks/templates/base.html" %}
+{% block footer %}
+{{ super() }}
+<script>
+document.addEventListener("DOMContentLoaded", inject);
+
+function inject() {
+  document.querySelectorAll(".code-selector").forEach(injectTabs);
+}
+
+function injectTabs(codeSelector) {
+  const nav = codeSelector.querySelector("ul.nav");
+
+  let langs = [];
+  let tabs = [];
+  let currentLang = null;
+  codeSelector.childNodes.forEach((element) => {
+    let toAdd = null;
+
+    if (element.nodeType === Node.COMMENT_NODE) {
+      // if its a comment like <!-- go -->, then set the currentLang to go and add it to langs
+      // end will cancel the currentLang, so you can continue to write common documentation.
+      const lang = element.data.trim();
+      if (lang === "end") {
+        currentLang = null;
+      } else {
+        currentLang = lang;
+        toAdd = { lang, element };
+      }
+    } else if (element.tagName === "PRE") {
+      // pre with data-lang attribute
+      const lang = element.getAttribute("data-lang");
+      // code blocks reset the currentLang.
+      currentLang = null;
+      toAdd = { lang, element };
+    } else if (currentLang) {
+      // if we are in a currentLang, then add the element to it.
+      toAdd = { lang: currentLang, element };
+    }
+
+    if (!toAdd) {
+      return;
+    }
+
+    langs.push(toAdd);
+    if (tabs.indexOf(toAdd.lang) === -1) {
+      tabs.push(toAdd.lang);
+    }
+  });
+
+  // Let us put this in the codeSelector element, so that we later check if the new selected language is in the list first.
+  codeSelector.langs = langs;
+
+  let saved = localStorage.getItem("code-lang");
+  let selected = tabs[0];
+  if (saved) {
+    // Only select the saved language if it's in the list.
+    found = tabs.find((lang) => lang === saved);
+    if (found) {
+      selected = found;
+    }
+  }
+
+  for (const lang of tabs) {
+    const isSelected = selected === lang;
```

Comment:
Could we rm this const and just use the nested expression in the if statement below, since it's only used there? Same for the `const isSelected` on L91

Reply:
I tend to like assigning variables to explain why an expression is made (instead of comments), but will fix this one since it's pretty obvious.